src/lime/extension/Main.hx

```diff
@@ -235,11 +235,15 @@ class Main
 		try
 		{
 			var output = ChildProcess.execSync(limeExecutable + " -version", {cwd: workspace.workspaceFolders[0].uri.fsPath});
-			limeVersion = StringTools.trim(Std.string(output));
+			var version = SemVer.parse(StringTools.trim(Std.string(output)));
+			if (version != null)
+			{
+				limeVersion = version;
+			}
 		}
 		catch (e:Dynamic)
```

Comment:
To me it looks like this try-catch was meant to preserve backwards compatibility, but `SemVer.parse()` never crashes, instead it returns `null`. So the NPE happens at a later point, completely preventing the extension from working with Lime versions that don't support `lime -version`.

Arguably this happened because of the implicit from-string-conversion that `SemVer` allowed.